### PR TITLE
Horizontal obstructed

### PR DESCRIPTION
**Pull request by EGimenoS**
Defined horizontal_obstructed? and tests, pretty much the same as vertical_obstructed.

### CODE REVIEW

**Comment by djacobs74** [comment]
Tests are assuming the board is empty, but the game appears to be being populated automatically by the game model itself in game.rb - 
after_create :initialize_board
 def initialize_board

if the top Rook on line 8 starting position is changed to 1, 2, and the first test is changed to 1, 5 the test passes.

Also it seems like the 'should return false if the horizontal move is not blocked moving from left to right of the board' test should fail if the board is being populated, since there should be pieces filling up X rows 0 and 1. 

Stephen and i looked at this and he suggested we could find a way to keep the board from being populated for testing purposes

**Comment by EGimenoS** [comment]
It looks like there are two ways to skip callbacks like initialize_board for tests.

1) write something like 
`describe Game do
  before(:all) do
    Game.any_instance.stub(:initialize_board)
  end`

2) Add this to the model to allow switching callbacks on and off

`class ActiveRecord::Base
  cattr_accessor :skip_callbacks
end`

Add this to the class Game:

`class Game < ActiveRecord::Base
  after_create :initialize_board, unless: :skip_callbacks
`
And this to tests
`ActiveRecord::Base.skip_callbacks = true
User.create # callbacks won't be run`

**Comment by EGimenoS** [comment]
@saakuffo @djacobs74 I set up Game factory so the callback initialize_board can be used or not depending on needs when creating a Game with FactoryGirl for test purposes. I also merged this branch with the latest master so it should be easier to merge. This branch also containts vertical_obstructed.

**Comment by mbrown78** [comment]
why are we passing in x in the some of your methods for example the def y_min method if it is not being used

File: app/models/piece.rb [hunk 1]
```diff
@@ -34,4 +34,45 @@ def horizontal?(x, y)
...
+  end
+
+  def x_max(x, y)
+    [x_position, x].max
+  end
+
+  # method to check is the piece move is vertically obstructed
+
+  def vertical_obstructed?(x, y)
+    (y_min(x, y)...y_max(x, y)).each do |y|
```
Comment (lasley) [inline app/models/piece.rb]: Why this outer loop? Wouldn't the SQL query suffice using the min/max?

File: app/models/piece.rb [hunk 2]
```diff
@@ -34,4 +34,45 @@ def horizontal?(x, y)
...
+        if (piece.x_position == x && piece.y_position == y) && piece.id != id
+          return true
+        end
+      end
+    end
+    false
+  end
+
+def horizontal_obstructed?(x, y)
+   (x_min(x, y)...x_max(x, y)).each do |x|
```
Comment (lasley) [inline app/models/piece.rb]: Why this outer loop? Wouldn't the SQL query suffice using the min/max?

File: app/models/piece.rb [hunk 3]
```diff
@@ -34,4 +34,45 @@ def horizontal?(x, y)
...
+  def x_max(x, y)
+    [x_position, x].max
+  end
+
+  # method to check is the piece move is vertically obstructed
+
+  def vertical_obstructed?(x, y)
+    (y_min(x, y)...y_max(x, y)).each do |y|
+      game.pieces.where('x_position = ? and y_position >= ? and y_position <= ?', x, y_min(x, y), y_max(x, y)).each do |piece|
+        if (piece.x_position == x && piece.y_position == y) && piece.id != id
```
Comment (lasley) [inline app/models/piece.rb]: Why double checking positions here? This logic is covered in the query it seems. 

Additionally, you could add `and id != ?` into the query to cover the id part of this

File: app/models/piece.rb [hunk 4]
```diff
@@ -34,4 +34,45 @@ def horizontal?(x, y)
...
+        end
+      end
+    end
+    false
+  end
+
+def horizontal_obstructed?(x, y)
+   (x_min(x, y)...x_max(x, y)).each do |x|
+      game.pieces.where('y_position = ? and x_position >= ? and x_position <= ?', y, x_min(x, y), x_max(x, y)).each do |piece|
+        if (piece.y_position == y && piece.x_position == x) && piece.id != id
```
Comment (lasley) [inline app/models/piece.rb]: Why double checking positions here? This logic is covered in the query it seems. 

Additionally, you could add `and id != ?` into the query to cover the id part of this

**Comment by EGimenoS** [comment]
I will review and submit in a PR that includes the whole is_obstructed method.

**Pull request closed without merging**